### ISSUE TITLE
BUG: Adding new columns doesn't work ass expected

### ISSUE THREAD

**Issue by AmT42** [modin-project/modin#5132]
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd
test = pd.DataFrame({"a":[1,2,4],"b":[0,17,5]})
|    |   a |   b |
|---:|----:|----:|
|  0 |   1 |   0 |
|  1 |   2 |  17 |
|  2 |   4 |   5 |


test.loc[test["a"].isin([1,4]),"c"] = np.nan
|    |   a |   b |
|---:|----:|----:|
|  0 |   1 | nan |
|  1 |   2 |  17 |
|  2 |   4 | nan |
```

### Issue Description

Hello,

I don't know if I m doing it wrong but when I try to create a new columns with a filter condition using loc it doesn't work.
It return me the dataframe with the last column changed.

### Expected Behavior

I expected something like this : 

|    |   a |   b |   c |
|---:|----:|----:|----:|
|  0 |   1 |   0 | nan |
|  1 |   2 |  17 | nan |
|  2 |   4 |   5 | nan |

### Error Logs

<details>

```python-traceback

Replace this line with the error backtrace (if applicable).

```

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 66e3805b8cabe977f40c05259cc3fcf7ead5687d
python           : 3.7.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-1080-aws
Version          : #87~18.04.1-Ubuntu SMP Fri Jun 10 18:32:41 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.5
numpy            : 1.21.6
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 22.3
setuptools       : 47.3.1.post20200622
Cython           : 0.29.20
pytest           : 5.4.3
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : None
pymysql          : 0.10.1
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.16.1
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.2
fsspec           : 2021.09.0
fastparquet      : 0.4.1
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : 1.3.20
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : 0.15.0
xlrd             : 1.2.0
xlwt             : None
numba            : 0.51.2

</details>

**Comment by mvashishtha** [comment]
Duplicate of #3764

**Comment by mvashishtha** [comment]
@AmT42 thank you for reporting this issue. The root cause is actually a known issue, #3764: assigning to a nonexistent column with `loc` should add that column to the dataframe. In this case column `"c"` doesn't exist in `test`, but assigning to it with `test.loc[test["a"].isin([1,4]),"c"] = np.nan` should create `c` and assign `np.nan` to it. I've marked this issue as a duplicate of #3764, so I'll close this one.

#3764 is a really important issue for Modin and we are actively working on a fix, #3765. Please follow progress on the thread for #3764.
